src/util/logging.py

```diff
@@ -17,30 +17,44 @@
 
 from util.context import Context
 
+log_call_count = dict()
+
+
+def log_multiline_message(message: Text, caller: Text):
+
+    # display logs called from one spot no more than 10 times, "None" means it was counted before
+    if caller != "None":
+        if log_call_count.get(caller, 0) < 10:
+            log_call_count[caller] = log_call_count.get(caller, 0) + 1
+        else:
+            return
 
-def log_multiline_message(message: Text):
     # need to modify endline char to have multiline log record not split into multiple log entries in CloudWatch:
     message = message.replace('\n', ' ')
     print(message)
 
 
-def debug_log_multiline_message(message: Text, context: Context):
+def debug_log_multiline_message(message: Text, context: Context, caller: Text):
     if context.debug:
-        log_multiline_message("DEBUG: " + message)
+        log_multiline_message("DEBUG: " + message, caller)
 
 
-def log_error_with_stacktrace(e: Exception, msg):
-    log_multiline_message(f"Exception '{e}' occurred. Additional message: '{msg}'")
-    log_multiline_message(traceback.format_exc())
+def log_error_with_stacktrace(e: Exception, msg, caller: Text):
+    if log_call_count.get(caller, 0) < 10:
```

Comment:
Since you are going to log_multiline_message anyway then this checks can and should stay in one place, in log_multiline_message. You can pass caller umodified. 
Since you are calling log_multiline_message twice and we don't want to break the throttling counter you could
log_multiline_message (..... , caller)
log_multiline_message (....., caller+"traceback")

Reply:
Maybe I could add an additional parameter to the function, like def log_multiline_message(message: Text, caller: Text, counted: bool = false) and set it to true the second time? We could be sure there would be no overlap with a function name

Reply:
Both will work fine but I find method signature you proposed a little non-obvious, hard to understand if you don't know this context. If I imagine myself being a person new to this and calling log_multiline_message for some reason, "what the hell is counted = bool ?? is this whether I want to apply throttling counting to my log?". But if you feel that counted=false is better then I am ok with this